src/com/amazon/ion/impl/_Private_IonTextAppender.java

```diff
@@ -712,7 +715,12 @@ public void printInt(BigInteger value)
             return;
         }
 
-        appendAscii(value.toString());
+        if (value.compareTo(MAX_LONG_BIG_INTEGER) > 0 || value.compareTo(MIN_LONG_BIG_INTEGER) < 0) {
```

Comment:
How does this compare?

```suggestion
        if (value.bitLength() >= Long.SIZE_BITS) {
```

Reply:
`value.bitLength() >= 64` is definitely clearer, and semantics are the same. I'll push a change for both affected lines :-)

Reply:
I snagged your JMH benchmark and ran the comparison, I didn't see much of a difference so I'd be happy with either. 

I'm going to run the benchmarks again with a bit more time to smooth them out (I got impatient and cut down the running time).

What would be great added here is a comment explaining the value of this optimization or referring to this PR where someone could read a more in-depth discussion of the trade-offs and alternatives considered. I think of that sort of comment as protecting the optimization from posterity :)